GoogleChromeLabs/ps-analysis-tool
# Feature: Optimise cookie updating iterations to reduce the cookie loading time.

## Description
This PR aims to considerably make the PSAT cookie data load faster and decrease the cookie display time.

## Relevant Technical Choices
- [x] Replace `chrome.storage.local` as a default way to store the cookie data.
- [x] Replace storage change listeners in both DevTools and Popup with Message Passing API.
- [x] Store all the cookie data in a cache manager.
- [x] Port all the methods used for cookie data in the cache manager.
- [x]  Reduce async calls in the service worker.
- [x] Remove Promise Queue from service worker.

## Testing Instructions

- Clone this branch. In the terminal run `npm start`.
- Open `edition.cnn.com` you should now see that the site cookies load faster than previous versions of extensions.

## Additional Information:



## Screenshot/Screencast



---

## Checklist






Fixes #380
Fixes #403
Fixes #404
Fixes #405
